huaweicloud/services/css/resource_huaweicloud_css_cluster.go

```diff
@@ -1670,6 +1700,131 @@ func updateCssTags(cssV1Client *cssv1.CssClient, id string, old, new map[string]
+	if err != nil {
+		return fmt.Errorf("error creating CSS V1 client: %s", err)
+	}
+	hcCssV1Client, err := conf.HcCssV1Client(region)
+	if err != nil {
+		return fmt.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	flavorId, err := flattenFlavorId(nodeType, d, resp)
+	if err != nil {
+		return err
+	}
+
+	updateFlavorHttpUrl := "v1.0/{project_id}/clusters/{cluster_id}/{types}/flavor"
+	updateFlavorPath := cssV1Client.Endpoint + updateFlavorHttpUrl
+	updateFlavorPath = strings.ReplaceAll(updateFlavorPath, "{project_id}", cssV1Client.ProjectID)
+	updateFlavorPath = strings.ReplaceAll(updateFlavorPath, "{cluster_id}", d.Id())
+	updateFlavorPath = strings.ReplaceAll(updateFlavorPath, "{types}", nodeType)
+
+	updateFlavorOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	updateFlavorOpt.JSONBody = map[string]interface{}{
+		"needCheckReplica": true,
+		"newFlavorId":      flavorId,
+		"isAutoPay":        1,
+	}
+	updateFlavorResp, err := cssV1Client.Request("POST", updateFlavorPath, &updateFlavorOpt)
+	if err != nil {
+		return fmt.Errorf("error updating CSS cluster flavor, cluster_id: %s, error: %s", d.Id(), err)
+	}
+
+	updateFlavorRespBody, err := utils.FlattenResponse(updateFlavorResp)
+	if err != nil {
+		return fmt.Errorf("error retrieving CSS cluster updating flavor response: %s", err)
+	}
+
+	orderId := utils.PathSearch("orderId", updateFlavorRespBody, "").(string)
+	if orderId != "" {
+		bssClient, err := conf.BssV2Client(region)
+		if err != nil {
+			return fmt.Errorf("error creating BSS v2 client: %s", err)
+		}
+
+		// 1. If charging mode is PrePaid, wait for the order to be completed.
+		err = common.WaitOrderComplete(ctx, bssClient, orderId, d.Timeout(schema.TimeoutUpdate))
+		if err != nil {
+			return err
+		}
+	}
+
+	err = checkClusterOperationCompleted(ctx, hcCssV1Client, d.Id(), d.Timeout(schema.TimeoutUpdate))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func flattenFlavorId(nodeType string, d *schema.ResourceData, resp *model.ListFlavorsResponse) (string, error) {
+	version := d.Get("engine_version").(string)
+	var flavorName string
+	switch nodeType {
+	case InstanceTypeEss:
+		flavorName = d.Get("ess_node_config.0.flavor").(string)
+	case InstanceTypeEssCold:
+		flavorName = d.Get("cold_node_config.0.flavor").(string)
+	case InstanceTypeEssMaster:
+		flavorName = d.Get("master_node_config.0.flavor").(string)
+	case InstanceTypeEssClient:
+		flavorName = d.Get("master_node_config.0.flavor").(string)
+	}
+	findFlavorIdChar := fmt.Sprintf(
+		`versions|[?type=='%s'&&version=='%s']|[0]|flavors|[?name=='%s']|[0]|flavor_id`,
+		nodeType, version, flavorName)
+	getFlavorsRespJson, err := json.Marshal(resp)
+	if err != nil {
+		return "", err
+	}
+	var data map[string]interface{}
+	err = json.Unmarshal(getFlavorsRespJson, &data)
+	if err != nil {
+		return "", err
+	}
+	flavorId, err := jmespath.Search(findFlavorIdChar, data)
```

Comment:
this err will always be nil, should use utils.PathSearch, and check whether the result is nil